### ISSUE TITLE
NPE in DefaultRecordMapper when returning it from custom RecordMapperProvider

### ISSUE THREAD

**Issue by artemSpb96** [jOOQ/jOOQ#7672]
Code from [JOOQ manual](https://www.jooq.org/doc/3.11/manual/sql-execution/fetching/pojos-with-recordmapper-provider/) cause NullPointerException

The problem is in line 

```java
boolean mapConstructorParameterNames = TRUE.equals(configuration.settings().isMapConstructorParameterNames());
```

in `init` method in `DefaultRecordMapper.`

When use custom RecordMapperProvider and create instance of DefaultRecordMapper then `init` method is called with `configuration = null`.

JOOQ version : 3.11

**Comment by lukaseder** [comment]
Thank you very much for your report. I can reproduce this with various other examples. Will fix immediately

**Comment by lukaseder** [comment]
Related: #7675

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. Backport scheduled for 3.11.4 (#7676) and 3.10.9 (#7677)
